esx_service/utils/auth_data.py

```diff
@@ -1157,6 +1157,27 @@ def remove_volumes_from_volumes_table(self, tenant_id):
 
         return None
 
+    def remove_symlink_for_tenant(self, tenant_id):
+        """
+            Delete the symlink /vmfs/volume/datastore_name/tenant_name
+            which point to /vmfs/volumes/datastore_name/tenant_uuid
+        """
+        err, tenant_name = self.get_tenant_name(tenant_id)
+        if err:
+            return err
+        VOL_RM_LOG_PREFIX = "Tenant <name> %s removal: "
+        for (datastore, _, path) in vmdk_utils.get_datastores():
```

Comment:
I don't understand the first part. What is the problem to loop over all datastores? I don't expect to have many datastores. How can we figure out the datastores assigned to the vmgroup without loop through all datastores? Could you explain more?

For the second part, yes,  I can add a test case for that.